src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,192 @@
+   */
+  onNodeExpanded?: (nodeId: string, expanded: boolean) => void;
+  /**
+   * Callback fired when selecting a TreeNode.
+   */
+  onNodeSelected?: (nodeId: string, selected: boolean) => void;
+  /**
+   * Checkbox to be shown before TreeNode.
+   * If undefined checkbox will not be shown.
+   * Recommended to use Checkbox component.
+   */
+  nodeCheckbox?: React.ReactNode;
+  /**
+   * Content shown after TreeNode.
+   */
+  children?: React.ReactNode;
+} & CommonProps;
+
+/**
+ * TreeNode component to display within a Tree.
+ * @example
+  <TreeNode
+    nodeId={props.nodeId}
+    label={props.node.label}
+    sublabel={props.node.subLabel}
+    onNodeExpanded={onNodeExpanded}
+    onNodeSelected={onSelectedNodeChange}
+    isDisabled={props.isDisabled}
+    isExpanded={props.isExpanded}
+    isSelected={props.isSelected}
+    nodeCheckbox={
+      <Checkbox variant='eyeball' disabled={props.isDisabled} />
+    }
+    icon={<SvgPlaceholder />}
+  />
+ */
+export const TreeNode = (props: TreeNodeProps) => {
+  const {
+    nodeId,
+    label,
+    sublabel,
+    children,
+    style,
+    className,
+    icon,
+    isDisabled = false,
+    isExpanded = false,
+    isSelected = false,
+    onNodeSelected,
+    onNodeExpanded,
+    nodeCheckbox,
+    ...rest
+  } = props;
+  useTheme();
+
+  const context = React.useContext(TreeContext);
+  const nodeDepth = context?.nodeDepth ?? 0;
+  const subNodeIds = context?.subNodeIds ?? [];
+
+  const styleLevel = React.useMemo(
+    () =>
+      getWindow()?.CSS?.supports?.(`--level: ${nodeDepth}`)
+        ? { '--level': nodeDepth, ...style }
+        : { marginLeft: nodeDepth ? nodeDepth * 28 : 0, ...style },
+    [nodeDepth, style],
+  );
+
+  const onNodeClick = () => {
+    if (isDisabled) {
+      return;
+    }
+    onNodeSelected?.(nodeId, !isSelected);
+  };
+
+  return (
+    <>
+      <li role='treeitem' id={nodeId} aria-expanded={isExpanded} {...rest}>
+        {
+          <div
+            className={cx(
+              'iui-tree-node',
+              {
+                'iui-active': isSelected,
+                'iui-disabled': isDisabled,
+              },
+              className,
+            )}
+            style={styleLevel}
+            onClick={() => onNodeClick()}
+            tabIndex={isSelected ? 0 : -1}
+          >
+            {nodeCheckbox && React.isValidElement(nodeCheckbox)
+              ? React.cloneElement(nodeCheckbox, {
+                  className: cx(
+                    'iui-tree-node-checkbox',
+                    nodeCheckbox.props.className,
+                  ),
+                })
+              : nodeCheckbox}
+            <div className='iui-tree-node-content'>
+              {context?.hasSubNodes && (
+                <IconButton
+                  styleType='borderless'
+                  size='small'
+                  onClick={(e) => {
+                    onNodeExpanded?.(nodeId, !isExpanded);
+                    e.stopPropagation();
```

Comment:
what it fixes?

Reply:
It prevents calling `onNodeClick` and selecting a node when expanding/collapsing a node,